### DISCUSSION TITLE
Open generic map configs with IncludeAllDerived

### DISCUSSION THREAD

**Discussion by engenb** [LuckyPennySoftware/AutoMapper#4037] [NA]
I've got a very contrived, hopefully simplified, example of my project:

```csharp
public interface ITrackEntityCreated
{
    DateTimeOffset Created { get; set; }
}

public interface ITrackEntityUpdated
{
    DateTimeOffset? Updated { get; set; }
}

public class User: ITrackEntityCreated, ITrackEntityUpdated
{
    public DateTimeOffset Created { get; set; }
    public DateTimeOffset? Updated { get; set; }
}
```

both `Created` and `Updated` properties are managed by the database, so I don't ever want AutoMapper to set these values, so:
```csharp
public class TrackEntityMaps: Profile
{
    public TrackEntityMaps()
    {
        CreateMap<object, ITrackEntityCreated>()
            .IncludeAllDerived()
            .ForMember(dest => dest.Created, opts => opts.Ignore());

        CreateMap<object, ITrackEntityUpdated>()
            .IncludeAllDerived()
            .ForMember(dest => dest.Updated, opts => opts.Ignore());
    }
}
```

I've found some behavior I don't understand with the following configuration.  I was expecting my open generic map configuration to inherit the configuration above because both my source and destination derive from `object` and `ITrackEntity[Created|Updated]`, respectively.

````csharp
public class UserCommandMaps : Profile
{
    public UserCommandMaps()
    {
        // this works.  it inherits the mapping configs defined in TrackEntityMaps
        CreateMap<UpdateUserCommand<Model.V1.User>, User>();
        CreateMap<UpdateUserCommand<Model.V2.User>, User>();

        // this doesn't inherit
        CreateMap(typeof(UpdateUserCommand<>), typeof(User));

        // this does "inherit" explicitly
        CreateMap(typeof(UpdateUserCommand<>), typeof(User))
            .IncludeBase(typeof(object), typeof(ITrackEntityCreated))
            .IncludeBase(typeof(object), typeof(ITrackEntityUpdated));
    }
}
````
Is this expected behavior or does this surprise anyone?

**Comment by lbargaoanu** [comment]
I guess it should work. A PR is welcome. You can target my branch as it's way ahead of master. I, for one, prefer the explicit version :)

**Comment by lbargaoanu** [comment]
It seems you're using an older version. Try the [MyGet](https://docs.automapper.org/en/latest/The-MyGet-build.html) build and go downward.

**Reply by engenb** [reply]
yeah sorry for not indicating that.  I'm using AutoMapper 11.0.1, which I believe is the latest stable release.  I'll try out a development build in the morning.
